### PR TITLE
[Javascript] Nullability: improve concept exercises

### PR DESCRIPTION
**Pull request by Jlamon**
Pull request for this issue: #1883 

There is an eslint error on nullability.js saying that the function's vars are defined but never used. Tell me if I should add an /* eslint-disable */

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Thank you for this! I will review it as soon as possible. However, since these are textual, I've invited @exercism/reviewers as well.

> There is an eslint error on nullability.js saying that the function's vars are defined but never used. 

That makes sense! They're unused 👍 . The student will fill that stub in, and then it will no longer be unused.

File: languages/javascript/exercises/concept/nullability/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+In this exercise, you'll be determining the text that needs to be printed on a badge.
+
+A badge requires the `id` of the employee, the `name` of the employee, as well as the department in which he is working.
+
+## 1. Text to print on the badge
+
+Implement a function that returns the text to print on the badge.
+
+Beware that the `id` should be between square brackets (i.e. `[20]`), and that only the id and department can be null.
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/instructions.md:9]:
```suggestion
The employee's name is always provided, but the employee `id` number and department name may be `null`. The employee `id` number, if provided, should be between square brackets (e.g. `[20]`).
```

File: languages/javascript/exercises/concept/nullability/.docs/instructions.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+A new owner should have no `id`, and no `department`
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/instructions.md:11]:
```suggestion
An owner's badge should not have an `id` or `department`
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Nullability in Javascript expresses the intentional absence of any object value. It is represented by the `null` value.
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:1]:
```suggestion
## nullability

The `null` value in JavaScript represents the intentional absence of an object value. It is one of JavaScript's primitive types.
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+```javascript
+// I do not have an apple.
+var apple = null;
+apple; // null
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:6]:
```suggestion
apple; // => null
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+// null is treated as falsy for boolean operations, therefore
+!apple; //true
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:9]:
```suggestion
!apple; // => true
!!apple; // => false
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+```
+
+`null` is part of Javascript's primitive values, and it has to be differentiated from the variable `undefined`. The
+distinction being that `undefined` is the value of an uninitialized variable or type, while `null` represents a missing object.
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:13]: I think this point is great, but probably belongs in the `about.md` document (formerly the `after.md` document).

```suggestion
Null values are often confused with `undefined` values.  They are two distinct primitive values in JavaScript. The distinction being that `undefined` is the value of an uninitialized variable or type, while `null` represents the absence of an object.
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+```javascript
+typeof null // object
+typeof undefined // undefined
+null === undefined // false
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:18]:
```suggestion
null === undefined // => evaluates to false by identity comparison
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+null == undefined // true
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:19]:
```suggestion
null == undefined // => evaluates to true by truthy comparison
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+// null is treated as falsy for boolean operations, therefore
+!apple; //true
+```
+
+`null` is part of Javascript's primitive values, and it has to be differentiated from the variable `undefined`. The
+distinction being that `undefined` is the value of an uninitialized variable or type, while `null` represents a missing object.
+
+```javascript
+typeof null // object
+typeof undefined // undefined
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:17]:
```suggestion
typeof null // => "object"
typeof undefined // => "undefined"
```

**Review by neenjaw — APPROVED** [review]
Hi @Jlamon! I'm part of the @exercism/reviewers team, and you've done a great job here! Don't be overwhelmed by my comments, I'm building on the great work you've done to start this off.

In the `.meta` folder, there is a `config.json`, make sure to add yourself as a contributor to get credit for your hard work!

---

In case you are not sure how reviews work here, we optimistically merge here, so I am going to approve your PR because I think it is a great start, but the comments I've left should probably either be addressed or put into an improvement issue if you don't want to (or don't have time to) incorporate them.

File: languages/javascript/concepts/nullability/about.md [hunk 10]
```diff
@@ -1 +1,70 @@
-TODO: add information on nullability concept
+## Nullability
+
+The key thing to remember about Null is that it is a primitive value, used to represent an intentional absence of an
```
Comment (ErikSchierboom) [inline languages/javascript/concepts/nullability/about.md:3]: As the concept documents are now standalone and no longer part of the exercise, maybe rephrase this to:

```suggestion
Null is a primitive value, used to represent an intentional absence of an
```

File: languages/javascript/exercises/concept/nullability/.docs/hints.md [hunk 11]
```diff
@@ -0,0 +1,4 @@
+## 1. Create the badge text.
+
+- Be careful about the differences between equality (`==`) and identity (`===`) operators.
+- Strings in JavaScript have several methods to change between upper and lower cases.
```
Comment (ErikSchierboom) [inline languages/javascript/exercises/concept/nullability/.docs/hints.md:4]: You could consider linking to a resource for the upper and/or lower case conversion.

File: languages/javascript/exercises/concept/nullability/.docs/instructions.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
+While working for a factory, a need arises to create the printed message on employee badges. You decide to implement
+this function in JavaScript.
+
+A badge requires the `id` of the employee, the `name` of the employee, as well as the department in which they are working.
+
+## 1. Create the badge text.
+
+Implement a function that returns the text to print on the badge.
```
Comment (ErikSchierboom) [inline languages/javascript/exercises/concept/nullability/.docs/instructions.md:8]:
```suggestion
Implement a function named `printBadge` that returns the text to print on the badge.
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+## Nullability
```
Comment (ErikSchierboom) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:1]: This should match the casing used in the config.json file (it will be replaced with the concept name when displayed):

```suggestion
## nullability
```

**Review by ErikSchierboom — APPROVED** [review]
Great work! I've left some comments.

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md (hunk 13 above)
Comment (Jlamon) [reply languages/javascript/exercises/concept/nullability/.docs/introduction.md:1]: Should change the title in `about.md` as well ?

Comment (ErikSchierboom) [reply languages/javascript/exercises/concept/nullability/.docs/introduction.md]: The `about.md` file of the concept should not have a heading.

Comment (Jlamon) [reply languages/javascript/exercises/concept/nullability/.docs/introduction.md]: Ok, it has been removed

File: languages/javascript/concepts/nullability/about.md [hunk 14]
```diff
@@ -1 +1,67 @@
-TODO: add information on nullability concept
+Null is a primitive value, used to represent an intentional absence of an object value.
+
+```javascript
+var person = null;
+
+if (person) {
+    console.log("Person is not null");
```
Comment (SleeplessByte) [inline languages/javascript/concepts/nullability/about.md:7]:
```suggestion
  console.log("person is not null");
```

File: languages/javascript/concepts/nullability/about.md [hunk 15]
```diff
@@ -1 +1,67 @@
...
+} else {
+    console.log("Person is null");
```
Comment (SleeplessByte) [inline languages/javascript/concepts/nullability/about.md:9]:
```suggestion
  console.log("person is null");
```

File: languages/javascript/concepts/nullability/about.md [hunk 16]
```diff
@@ -1 +1,67 @@
-TODO: add information on nullability concept
+Null is a primitive value, used to represent an intentional absence of an object value.
```
Comment (SleeplessByte) [inline languages/javascript/concepts/nullability/about.md:1]:
```suggestion
Null is a primitive value, used to represent an intentional absence of an object value. It's _falsy_, and its `typeof` is `object`.
```

File: languages/javascript/concepts/nullability/about.md [hunk 17]
```diff
@@ -1 +1,67 @@
...
+
+## Null and truthy comparisons
+
+Be aware on how `null` works with the equality (`==`) and identity (`===`) operators.
+
+```javascript
+null === null // => evaluates to true by identity comparison
+null == null // => evaluates to false by truthy comparison
+```
+
```
Comment (SleeplessByte) [inline languages/javascript/concepts/nullability/about.md:35]:
```suggestion
```

This is not correct.

File: languages/javascript/exercises/concept/nullability/.docs/instructions.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
+While working for a factory, a need arises to create the printed message on employee badges. You decide to implement
+this function in JavaScript.
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/nullability/.docs/instructions.md:2]:
```suggestion
While working for a factory, a need arises to create the printed message on employee badges. You decide to implement this function in JavaScript.
```

File: languages/javascript/exercises/concept/nullability/.docs/instructions.md [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+A badge requires the `id` of the employee, the `name` of the employee, as well as the department in which they are working.
+
+## 1. Create the badge text.
+
+Implement a function named `printBadge` that returns the text to print on the badge.
+
+The employee's name is always provided, but the employee `id` number and department name may be `null`. The employee
+`id` number, if provided, should be between square brackets (e.g. `[20]`).
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/nullability/.docs/instructions.md:11]:
```suggestion
The employee's name is always provided, but the employee `id` number and department name may be `null`. The employee `id` number, if provided, should be between square brackets (e.g. `[20]`).
```

File: languages/javascript/exercises/concept/nullability/.docs/introduction.md [hunk 20]
```diff
@@ -0,0 +1,14 @@
+## nullability
+
+The `null` value in Javascript represents the intentional absence of an object value. It is one of Javascript's primitive
+types.
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/nullability/.docs/introduction.md:4]:
```suggestion
The `null` value in Javascript represents the intentional absence of an object value. It is one of Javascript's primitive types.
```

**Review by SleeplessByte — APPROVED** [review]

**Comment by SleeplessByte** [comment]
Thanks a million @Jlamon . This was great work :)
